### PR TITLE
#28: Add type checking for the provided id parameters

### PR DESCRIPTION
**Pull request by RobLoach**
#28

### CODE REVIEW

**Comment by RobLoach** [comment]
Need to update `README.md`:
https://github.com/RobLoach/jquery-once/blob/28-argumenttype/README.md#usage

**Comment by RobLoach** [comment]
@theodoreb Feel free to merge/fix this however you want. Looks pretty good to me. The magic is the checkId function:
https://github.com/RobLoach/jquery-once/blob/28-argumenttype/jquery.once.js#L32

File: jquery.once.js [hunk 1]
```diff
@@ -51,37 +87,39 @@
...
-    }
-    // Build the name for the data identifier. Generate a new unique id if the
-    // id parameter is not provided.
-    var name = "jquery-once-" + id;
+    // Build the name for the data identifier. Generate a new unique ID if the
+    // ID parameter is not provided.
+    var name = "jquery-once-" + checkId(id);
 
-    // Filter the elements by which do not have the data yet.
+    // Filter the elements by whi*ch do not have the data yet.
```
Comment (theodoreb) [inline jquery.once.js]: Typo?

**Comment by theodoreb** [comment]
Looks good to me, how do you handle the minified file ? as part of PRs or do you update it when you're tagging a new release?

**Comment by RobLoach** [comment]
> Looks good to me, how do you handle the minified file ? as part of PRs or do you update it when you're tagging a new release?

I've been updating it with [the development notes](https://github.com/robloach/jquery-once#development) each time before tagging. Know of any easier workflow?

**Pull request merged**